src/cartesian/Area.tsx

```diff
@@ -593,29 +595,50 @@ export class Area extends PureComponent<Props, State> {
     const dotSize = r * 2 + strokeWidth;
 
     return (
-      <Layer className={layerClass}>
-        <SetAreaLegend {...this.props} />
-        {needClipX || needClipY ? (
-          <defs>
-            <clipPath id={`clipPath-${clipPathId}`}>
-              <rect
-                x={needClipX ? left : left - width / 2}
-                y={needClipY ? top : top - height / 2}
-                width={needClipX ? width : width * 2}
-                height={needClipY ? height : height * 2}
-              />
-            </clipPath>
-            {!clipDot && (
-              <clipPath id={`clipPath-dots-${clipPathId}`}>
-                <rect x={left - dotSize / 2} y={top - dotSize / 2} width={width + dotSize} height={height + dotSize} />
+      <>
+        <Layer className={layerClass}>
+          <SetAreaLegend {...this.props} />
+          {needClipX || needClipY ? (
+            <defs>
+              <clipPath id={`clipPath-${clipPathId}`}>
+                <rect
+                  x={needClipX ? left : left - width / 2}
+                  y={needClipY ? top : top - height / 2}
+                  width={needClipX ? width : width * 2}
+                  height={needClipY ? height : height * 2}
+                />
               </clipPath>
-            )}
-          </defs>
-        ) : null}
-        {!hasSinglePoint ? this.renderArea(needClip, clipPathId) : null}
-        {(dot || hasSinglePoint) && this.renderDots(needClip, clipDot, clipPathId)}
-        {(!isAnimationActive || isAnimationFinished) && LabelList.renderCallByParent(this.props, points)}
-      </Layer>
+              {!clipDot && (
+                <clipPath id={`clipPath-dots-${clipPathId}`}>
+                  <rect
+                    x={left - dotSize / 2}
+                    y={top - dotSize / 2}
+                    width={width + dotSize}
+                    height={height + dotSize}
+                  />
+                </clipPath>
+              )}
+            </defs>
+          ) : null}
+          {!hasSinglePoint ? this.renderArea(needClip, clipPathId) : null}
+          {(dot || hasSinglePoint) && this.renderDots(needClip, clipDot, clipPathId)}
+          {(!isAnimationActive || isAnimationFinished) && LabelList.renderCallByParent(this.props, points)}
+        </Layer>
+        <ActivePoints
+          points={points}
+          mainColor={getLegendItemColor(this.props.stroke, this.props.fill)}
+          itemDataKey={this.props.dataKey}
+          activeDot={this.props.activeDot}
+        />
+        {this.props.isRange && Array.isArray(baseLine) && (
```

Comment:
what is this check doing? We render two sets of active dots for area under certain conditions?

Reply:
Correct, Area shows two dots. You can see it in one of the storybooks I added recently.